### PR TITLE
AAE-51054 Fix the released task status prerequisite bug fix

### PR DESCRIPTION
**Pull request by elena-grosu-h**
Fixes how the query service updates task assignment state when consuming TASK_ASSIGNED events, addressing the “release” scenario where the assignee is cleared and the task should return to CREATED.

Changes:

- Update TaskAssignedEventHandler to set task status to ASSIGNED only when an assignee is present, otherwise set it to CREATED.

- Add a focused unit test suite covering claim/release/claim cycles and empty/null assignee cases.

- Add a new query feature toggle constant (FEATURE_PUSHED_COUNTS) for future pushed/pre-aggregated count support.

### CODE REVIEW

File: activiti-cloud-query-service/activiti-cloud-services-query/activiti-cloud-services-query-events/src/main/java/org/activiti/cloud/services/query/events/handlers/TaskAssignedEventHandler.java [hunk 1]
```diff
@@ -41,8 +41,13 @@ public void handle(CloudRuntimeEvent<?, ?> event) {
...
         );
-        queryTaskEntity.setAssignee(eventTask.getAssignee());
-        queryTaskEntity.setStatus(Task.TaskStatus.ASSIGNED);
+        String assignee = eventTask.getAssignee();
+        queryTaskEntity.setAssignee(assignee);
+        if (assignee != null && !assignee.isEmpty()) {
+            queryTaskEntity.setStatus(Task.TaskStatus.ASSIGNED);
+        } else {
+            queryTaskEntity.setStatus(Task.TaskStatus.CREATED);
+        }
```
[the remark this answers is not in the document]
Comment (elena-grosu-h) [reply activiti-cloud-query-service/activiti-cloud-services-query/activiti-cloud-services-query-events/src/main/java/org/activiti/cloud/services/query/events/handlers/TaskAssignedEventHandler.java]: Applied.

**Review by jsokolowskii — APPROVED** [review]
